src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -297,21 +295,37 @@ private void setupBatchHttpStream(Promise<?> promise, HttpRequestImpl<Buffer> in
       }
       downloadInstancesPromise.complete();
     });
-    jsonParser.exceptionHandler(throwable -> {
-      logger.error("Error has been occurred at JsonParser while reading data from response. Message: {}", throwable.getMessage(),
+    jsonParser.exceptionHandler(throwable -> responseChecked.future().onSuccess(invalidResponseReceivedAndProcessed -> {
+        if (invalidResponseReceivedAndProcessed) {
```

Comment:
I don't think that it is a code smell. I can add `invalidResponseReceivedAndProcessed.booleanValue()` if you think that this matters.